Sources/URL.swift

```diff
@@ -15,15 +15,16 @@ struct URL {
     }
 
     scheme = parts[0]
-    let parts1 = parts[1].split("/", maxSplit: 1)
-    if parts1.count != 2 {
+    let parts1 = parts[1].split("/")
+    if parts1.count < 2 {
       hostname = ""
       port = 80
       path = ""
+      print("foo")
       return nil
     }
 
-    path = "/" + parts1[1]
+    path = "/" + parts1[1..<parts1.count].joinWithSeparator("/")
```

Comment:
I think instead of separating and rejoining, best to separate with maxSplit 1, but add clause to else so that path defaults to `/`. Otherwise things like `/test///magic` would be re-joined incorrect I think.